File: 2. Add Two Numbers/Solution.cpp
#include <bits/stdc++.h>

using namespace std;

int faster = [](){
    ios::sync_with_stdio(0);
    cin.tie(0);
    return 0;
}();

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};
 
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* head = new ListNode(0);
        ListNode* next = head;
        int carry = 0;
        int val = 0;
        while (l1 || l2 || carry) {
            val = carry;
            if (l1) {
                val += l1->val;
                l1 = l1->next;
            }
            if (l2) {
                val += l2->val;
                l2 = l2->next;
            }
            carry = val / 10;
            val %= 10;
            next->next = new ListNode(val);
            next = next->next;
        }

        return head->next;
    }
};

File: 2. Add Two Numbers/Solution.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        ans = ListNode()
        last = ans
        l = 0
        while l1.next or l2.next:
            l = l + l1.val + l2.val
            last.val = l%10
            l = l//10
            last.next=ListNode()
            last = last.next
            l1 = l1.next if l1.next else ListNode()
            l2 = l2.next if l2.next else ListNode()
        l = l + l1.val + l2.val
        last.val = l%10
        l = l//10
        if l:
            last.next = ListNode(l)
        return ans
